File: libs/sdk-bindings/tests/bindings/golang/test_breez_sdk.go
package main

import (
	"fmt"
	"log"

	"example.org/golang/breez_sdk"
)

type BreezListener struct{}

func (BreezListener) Log(l breez_sdk.LogEntry) {
	if l.Level != "TRACE" {
		fmt.Printf("%v\n", l.Line)
	}
}

func (BreezListener) OnEvent(e breez_sdk.BreezEvent) {
	fmt.Printf("%#v", e)
}

func main() {
	breezListener := BreezListener{}

	filterLevel := breez_sdk.LevelFilterTrace

	breez_sdk.SetLogStream(breezListener, &filterLevel)

	seed, sdkErr := breez_sdk.MnemonicToSeed("repeat hawk combine screen network rhythm ritual social neither casual volcano powder")

	if sdkErr != nil {
		log.Fatalf("MnemonicToSeed failed: %#v", sdkErr)
	}

	config := breez_sdk.DefaultConfig(breez_sdk.EnvironmentTypeProduction, "code", breez_sdk.NodeConfigGreenlight{
		Config: breez_sdk.GreenlightNodeConfig{
			PartnerCredentials: nil,
			InviteCode:         nil,
		},
	})
	connectRequest := breez_sdk.ConnectRequest{Config: config, Seed: seed}
	sdkServices, connectErr := breez_sdk.Connect(connectRequest, breezListener)

	if connectErr != nil {
		log.Fatalf("Connect failed: %#v", connectErr)
	}

	nodeInfo, sdkErr := sdkServices.NodeInfo()

	if sdkErr != nil {
		log.Fatalf("NodeInfo failed: %#v", sdkErr)
	}

	log.Print(nodeInfo.Id)
}

File: libs/sdk-bindings/tests/bindings/test_breez_sdk.cs

using breez.breez_sdk;

try
{
 var seed = BreezSdkMethods.MnemonicToSeed("repeat hawk combine screen network rhythm ritual social neither casual volcano powder");
 BreezSdkMethods.SetLogStream(new LogStreamListener(), LevelFilter.TRACE);
 var config = BreezSdkMethods.DefaultConfig(EnvironmentType.PRODUCTION, "code", new NodeConfig.Greenlight(new GreenlightNodeConfig(null, null)));
 var connectRequest = new ConnectRequest(config, seed);
 BlockingBreezServices sdkServices = BreezSdkMethods.Connect(connectRequest, new SDKListener());
 NodeState? nodeInfo = sdkServices.NodeInfo();
 Console.WriteLine(nodeInfo!.id);
}
catch (Exception e)
{
 Console.WriteLine(e.Message);
 throw;
}

class SDKListener : EventListener
{
 public void OnEvent(BreezEvent e)
 {
  Console.WriteLine("received event " + e);
 }
}

class LogStreamListener : LogStream
{
 public void Log(LogEntry l)
 {
  if (l.level != "TRACE") {
   Console.WriteLine(l.line);
  }
 }
}
